cpp/src/hash/concurrent_unordered_map.cuh

```diff
@@ -263,7 +265,7 @@ class concurrent_unordered_map {
    */
   template <typename pair_type = value_type>
   __device__ std::enable_if_t<is_packable<pair_type>(), insert_result> attempt_insert(
-    value_type* insert_location, value_type const& insert_pair)
+    value_type* const __restrict__ insert_location, value_type const& insert_pair)
```

Comment:
is const relevant here?

Reply:
Not really. I just wanted to two `attempt_insert` functions defined here to match. And removing `const` from the 2nd one just seemed wrong.

Reply:
No harm in keeping it, I was just curious if there's maybe an interaction with `__restrict__`.